fmriprep/interfaces/resampling.py

```diff
@@ -0,0 +1,641 @@
+    File,
+    InputMultiObject,
+    SimpleInterface,
+    TraitedSpec,
+    traits,
+)
+from nipype.utils.filemanip import fname_presuffix
+from scipy import ndimage as ndi
+from scipy.sparse import hstack as sparse_hstack
+from sdcflows.transform import grid_bspline_weights
+from sdcflows.utils.tools import ensure_positive_cosines
+
+from ..utils.asynctools import worker
+from ..utils.transforms import load_transforms
+
+
+class ResampleSeriesInputSpec(TraitedSpec):
+    in_file = File(mandatory=True, desc="3D or 4D image file to resample")
+    ref_file = File(mandatory=True, desc="File to resample in_file to")
+    transforms = InputMultiObject(
+        File, mandatory=True, desc="Transform files, from in_file to ref_file (image mode)"
+    )
+    inverse = InputMultiObject(
+        traits.Bool,
+        value=[False],
+        usedefault=True,
+        desc="Whether to invert each file in transforms",
+    )
+    fieldmap = File(desc="Fieldmap file resampled into reference space")
+    ro_time = traits.Float(desc="EPI readout time (s).")
+    pe_dir = traits.Enum(
+        "i",
+        "i-",
+        "j",
+        "j-",
+        "k",
+        "k-",
+        desc="the phase-encoding direction corresponding to in_data",
+    )
+    num_threads = traits.Int(1, usedefault=True, desc="Number of threads to use for resampling")
+
+
+class ResampleSeriesOutputSpec(TraitedSpec):
+    out_file = File(desc="Resampled image or series")
+
+
+class ResampleSeries(SimpleInterface):
+    """Resample a time series, applying susceptibility and motion correction
+    simultaneously.
+    """
+
+    input_spec = ResampleSeriesInputSpec
+    output_spec = ResampleSeriesOutputSpec
+
+    def _run_interface(self, runtime):
+        out_path = fname_presuffix(self.inputs.in_file, suffix='resampled', newpath=runtime.cwd)
+
+        source = nb.load(self.inputs.in_file)
+        target = nb.load(self.inputs.ref_file)
+        fieldmap = nb.load(self.inputs.fieldmap) if self.inputs.fieldmap else None
+
+        nvols = source.shape[3] if source.ndim > 3 else 1
+
+        transforms = load_transforms(self.inputs.transforms, self.inputs.inverse)
+
+        pe_dir = self.inputs.pe_dir
+        ro_time = self.inputs.ro_time
+        pe_info = None
+
+        if pe_dir and ro_time:
+            pe_axis = "ijk".index(pe_dir[0])
+            pe_flip = pe_dir.endswith("-")
+
+            # Nitransforms displacements are positive
+            source, axcodes = ensure_positive_cosines(source)
+            axis_flip = axcodes[pe_axis] in "LPI"
+
+            pe_info = [(pe_axis, -ro_time if (axis_flip ^ pe_flip) else ro_time)] * nvols
+
+        resampled = resample_bold(
+            source=source,
+            target=target,
+            transforms=transforms,
+            fieldmap=fieldmap,
+            pe_info=pe_info,
+            nthreads=self.inputs.num_threads,
+        )
+        resampled.to_filename(out_path)
+
+        self._results = {'out_file': out_path}
```

Comment:
to make this extendable if future outputs are added
```suggestion
        self._results['out_file'] = out_path
```